mypy/renaming.py

```diff
@@ -275,36 +261,38 @@ def clear(self) -> None:
         self.blocks = []
         self.var_blocks = []
 
-    def enter_block(self) -> None:
+    @contextmanager
+    def enter_block(self) -> Iterator[None]:
         self.block_id += 1
         self.blocks.append(self.block_id)
         self.block_loop_depth[self.block_id] = self.loop_depth
+        try:
+            yield
+        finally:
+            self.blocks.pop()
 
-    def leave_block(self) -> None:
-        self.blocks.pop()
-
-    def enter_try(self) -> None:
+    @contextmanager
+    def enter_try(self) -> Iterator[None]:
         self.disallow_redef_depth += 1
-
-    def leave_try(self) -> None:
+        yield
```

Comment:
These also need try-finally, same for enter_scope

Reply:
Yep, just updated all of them to use a try-finally!